extensions/amp-story/1.0/amp-story-quiz.css

```diff
@@ -170,16 +192,69 @@ h3.i-amphtml-story-quiz-prompt {
     line-height: 20px !important;
 }
 
-.i-amphtml-story-quiz-post-selection .i-amphtml-story-quiz-option-selected[correct].i-amphtml-story-quiz-option::before {
+.i-amphtml-story-quiz-post-selection:not(.i-amphtml-story-quiz-has-data) .i-amphtml-story-quiz-option-selected[correct].i-amphtml-story-quiz-option::before {
     animation: option-select-correct forwards 150ms cubic-bezier(0.0, 0.0, 0.2, 1) !important;
     border: solid 1px !important;
 }
 
-.i-amphtml-story-quiz-post-selection .i-amphtml-story-quiz-option-selected:not([correct]).i-amphtml-story-quiz-option::before {
+.i-amphtml-story-quiz-post-selection:not(.i-amphtml-story-quiz-has-data) .i-amphtml-story-quiz-option-selected:not([correct]).i-amphtml-story-quiz-option::before {
     animation: option-select-incorrect forwards 150ms cubic-bezier(0.0, 0.0, 0.2, 1) !important;
     border: solid 1px !important;
 }
 
+.i-amphtml-story-quiz-post-selection.i-amphtml-story-quiz-has-data .i-amphtml-story-quiz-option:nth-of-type(1)::before {
+    transform: translateX(var(--option-1-percentage)) !important;
+}
+
+.i-amphtml-story-quiz-post-selection.i-amphtml-story-quiz-has-data .i-amphtml-story-quiz-option:nth-of-type(2)::before {
+    transform: translateX(var(--option-2-percentage)) !important;
+}
+
+.i-amphtml-story-quiz-post-selection.i-amphtml-story-quiz-has-data .i-amphtml-story-quiz-option:nth-of-type(3)::before {
+    transform: translateX(var(--option-3-percentage)) !important;
+}
+
+.i-amphtml-story-quiz-post-selection.i-amphtml-story-quiz-has-data .i-amphtml-story-quiz-option:nth-of-type(4)::before {
+    transform: translateX(var(--option-4-percentage)) !important;
+}
+
+.i-amphtml-story-quiz-post-selection.i-amphtml-story-quiz-has-data .i-amphtml-story-quiz-option::before {
+    left: -101% !important;
+    width: 100% !important;
+    height: 100% !important;
+    border-radius: 0px !important;
+    background: #ECEDEF !important;
+    color: #ECEDEF !important;
+    transition: transform 1s !important;
+    border: solid 2px !important;
+}
+
+.i-amphtml-story-quiz-post-selection.i-amphtml-story-quiz-has-data .i-amphtml-story-quiz-option-selected[correct].i-amphtml-story-quiz-option {
+    background: var(--correct-color) !important;
+    border: none !important;
+    /* Ups the padding by one to account for the border which must be removed
+     * to allow the shading to reach the edge of the box */
```

Comment:
Please look into this https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/The_box_model#What_is_the_CSS_box_model and the "alternative box model", that should help you simplifying this code :)